serializer/processor.go

```diff
@@ -0,0 +1,110 @@
+package serializer
+
+import (
+	"go.temporal.io/sdk/log"
+	"go.temporal.io/sdk/workflow"
+)
+
+type (
+	EventProcessor struct {
+		ctx     workflow.Context
+		logger  log.Logger
+		eventCh workflow.ReceiveChannel
+
+		state *EventProcessorState
+	}
+
+	EventProcessorState struct {
+		PreviousEventID int
+		Items           map[int]*ResourceEvent
```

Comment:
Consider not using pointers for small structures.

Reply:
Done.